docs/development/vscode.md

```diff
@@ -0,0 +1,143 @@
+helps them significantly ! :D
+
+## Steps
+
+1. Download and install VS Code.
+2. Download and install the
+   [Python](https://marketplace.visualstudio.com/items?itemName=ms-python.python)
+   extension for VS Code by Microsoft.  
+3. Set up the vagrant dev environment from [here](./setup-advanced.md)
+4. Make sure you run the dev server at least once. ( So that you know
+   everything is working before making changes to the files )
+5. Expose *port* in Vagrant file
+
+    Add the following line in `./Vagrantfile` and replace
+    `YOUR_GUEST_PORT_NUMBER` which you want to connect your VS Code
+    Debugger to the ptvsd server and `YOUR_HOST_PORT_NUMBER` to one on
+    which ptvsd server is going to run on.
+
+    **Note**: You can keep both same for simplicity
+    ```
+    config.vm.network "forwarded_port",
+    guest: YOUR_GUEST_PORT_NUMBER,
+    host: YOUR_HOST_PORT_NUMBER,
+    host_ip: host_ip_addr
+    ```
+    Example:
+    ```
+    config.vm.network "forwarded_port", guest: 5678, host: 5678, host_ip: host_ip_addr
+    ```
+
+6. Expose port in Docker
+
+    If you are using a *virtual box* instance, you can skip this step.
+    Add the following lines at the end of the
+    `tools/setup/dev-vagrant-docker` and replace
+    `YOUR_PTVSD_PORT_NUMBER` with the port number on which ptvsd server
+    will run. (This must be same as the `YOUR_HOST_PORT_NUMBER` from
+    last step)
+    ```docker
+    EXPOSE YOUR_PTVSD_PORT_NUMBER
+    ```
+    Example:
+
+    ```docker
+    EXPOSE 5678
+    ```
+
+7. Download and install ptvsd by `ssh-ing` into the vagrant server.
+
+    **Note**: You should activate the `zulip-py3-venv` if not already
+    activated and then run the following.
+
+    ```python
+    pip install ptvsd
+    ```
+
+8. Edit `tools/settings.py`
+
+    Add the following lines in `zproject/settings.py` after `if DEBUG:`
+    ```python
+    if DEBUG:                                   # Already present
+        INTERNAL_IPS = ('127.0.0.1',)           # Already Present
+        import ptvsd
+        if "runserver" in sys.argv and os.environ.get("RUN_MAIN"):
+            try:
+                ptvsd.enable_attach(redirect_output=True)
+                print("VS Code Debugging Enabled.")
+            except OSError:
+                print("Debugging Setup Error, Exception: ")
+                tracback.print_exc()
+    ```
+
+    The `redirect_output` gets the output from the server redirected to
+    the debug console of VS Code (quite a handy thing!).
+
+    Here you can have a bit of modifications like:
+    - If you want the django server to wait for debugger *prior to
+      running* just add
+    ```python
+    ...
+    ptvsd.enable_attach()
+    ptvsd.wait_for_debugger()
+    ...
+    ```
+   - If you want to run the ptvsd server on some other port than default
+     (default is *0.0.0.0* and *5678* ) then
+    ```python
+    ptvsd.enable_attach(YOUR_HOST_IP, YOUR_HOST_PORT)
+    ```
```

Comment:
We could put a basic version of this logic inside `zproject/dev_settings.py`, at the very least in an `if os.environ.get("VSCODE_DEBUG")`

Reply:
I would need some help regarding this. I would definitely love to try this.